src/main/java/org/reaktivity/nukleus/sse/internal/stream/ServerStreamFactory.java

```diff
@@ -814,6 +832,55 @@ private void handleReset(
             final long traceId = reset.trace();
             doReset(applicationReplyThrottle, applicationRouteId, applicationReplyId, traceId);
         }
+
+        private void handleChallenge(
+            ChallengeFW challenge)
+        {
+            final HttpChallengeExFW httpChallengeEx = challenge.extension().get(httpChallengeExRO::tryWrap);
+            if (httpChallengeEx != null)
+            {
+                final JsonObject jsonHeaders = new JsonObject();
+                final JsonObject challengeJson = new JsonObject();
+                final ListFW<HttpHeaderFW> httpHeaders = httpChallengeEx.headers();
+
+                httpHeaders.forEach(header ->
+                {
+                    final String name = header.name().asString();
+                    final String value = header.value().asString();
+                    if (name != null)
+                    {
+                        if (!name.startsWith(":"))
+                        {
+                            jsonHeaders.addProperty(name, value);
+                        }
+                        else if (name.equals(":method"))
+                        {
+                            challengeJson.addProperty("method", value);
+                        }
+                    }
+                });
+                challengeJson.addProperty("headers", gson.toJson(jsonHeaders));
+                writeBuffer.putStringUtf8(0, gson.toJson(challengeJson));
+                final StringFW challengeData = stringRO.wrap(writeBuffer, 0, writeBuffer.capacity());
+
+                final SseEventFW sseEvent = sseEventRW.wrap(writeBuffer, DataFW.FIELD_OFFSET_PAYLOAD, writeBuffer.capacity())
+                        .type(EVENT_TYPE_CHALLENGE.value())
+                        .data(challengeData)
+                        .build();
+
+                final DataFW data = dataRW.wrap(writeBuffer, 0, writeBuffer.capacity())
+                        .routeId(networkRouteId)
+                        .streamId(networkReplyId)
+                        .trace(challenge.trace())
+                        .authorization(challenge.authorization())
+                        .groupId(0)
+                        .padding(networkReplyPadding)
+                        .payload(sseEvent.buffer(), sseEvent.offset(), sseEvent.sizeof())
+                        .build();
```

Comment:
Sending this `DATA` frame also needs to account for flow control budget usage, as a window will be later returned to restore the budget, so if we don't subtract here, then the understanding of current budget will get out of sync.

So, that means we also need to deal with potentially insufficient budget to write the `DATA` frame immediately. In that case it will need to be buffered and flushed when the next window arrives if there is then sufficient budget, otherwise wait for another window.

Storage would be in a `bufferPool` slot, acquired when needed and released again when no longer required.